### PR TITLE
:bug: work with kind v0.7.0

### PR DESCRIPTION
**Pull request by trutx**
**What this PR does / why we need it**:

Updates package to work with newer versions of `kind`. There's been some deprecations and changes in `v0.7.0`, especially the removal of `kind get kubeconfig-path` subcommand which makes this package fail when creating clusters.

This PR also adds a few more cluster configuration options for better fine-tuning and granularity.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

NONE

### CODE REVIEW

**Comment by chuckha** [comment]
This is only used by CAPA and not used internally in Cluster API at all so I'm going to assign this to @detiber.

From my perspective, test/helpers is deprecated and all use should be replaced by the test framework that is more actively maintained or open issues if the framework is not sufficient.

/assign @detiber 
/unassign

File: test/helpers/kind/setup.go [hunk 1]
```diff
@@ -19,91 +19,119 @@ package kind
...
-	c.tmpDir, err = ioutil.TempDir("", "kind-home")
-	ExpectWithOffset(1, err).NotTo(HaveOccurred())
-	fmt.Fprintf(GinkgoWriter, "creating Kind cluster named %q\n", c.Name)
-	c.run(exec.Command(*kindBinary, "create", "cluster", "--image", kindestImage, "--name", c.Name))
-	path := c.runWithOutput(exec.Command(*kindBinary, "get", "kubeconfig-path", "--name", c.Name))
-	c.kubepath = strings.TrimSpace(string(path))
-	fmt.Fprintf(GinkgoWriter, "kubeconfig path: %q. Can use the following to access the cluster:\n", c.kubepath)
-	fmt.Fprintf(GinkgoWriter, "export KUBECONFIG=%s\n", c.kubepath)
+	c.kindBinary, err = exec.LookPath("kind")
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
```
Comment (fabriziopandini) [inline test/helpers/kind/setup.go]: Does it makes sense to use ginko inside helpers/framework in light of https://github.com/kubernetes-sigs/cluster-api/issues/2306?
@detiber @chuckha ^^

Comment (detiber) [reply test/helpers/kind/setup.go]: These helpers will eventually be going away in favor of the e2e test framework.

File: test/helpers/kind/setup.go [hunk 2]
```diff
@@ -19,91 +19,119 @@ package kind
...
-	"io/ioutil"
 	"os"
 	"os/exec"
-	"strings"
 	"sync"
 
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/gomega"
-
+	"github.com/onsi/ginkgo"
```
[the remark this answers is not in the document]
Comment (vincepri) [reply test/helpers/kind/setup.go]: We shouldn't imo, dot imports improve readability a lot and I'm +1 using them everywhere we can

**Comment by vincepri** [comment]
/ok-to-test
/hold
Holding until after v0.3.0 is cut

**Comment by detiber** [comment]
I'm not currently aware of anywhere this is used outside of cluster-api-provider-aws. If that is the case, then I think I would rather prioritize the migration of cluster-api-provider-aws to the newer e2e framework and work towards removal of this helper.

File: test/helpers/kind/setup.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (trutx) [reply test/helpers/kind/setup.go]: Not really, my linter suggested not to use them so I went ahead but honestly I also like them better. Fixing in a sec.

**Comment by vincepri** [comment]
If this is only being used in CAPA and it's effectively deprecated, we probably should remove it from CAPI and move it there until we can remove it

**Comment by vincepri** [comment]
If we want to merge this in, let's add a big warning sign that it's deprecated and folks shouldn't use it ⚠️

**Comment by trutx** [comment]
So you mean adding an explicit comment at the top of the file?

**Comment by vincepri** [comment]
@detiber would you want to move these files to CAPA instead of keeping them here?

**Comment by vincepri** [comment]
@trutx After chatting with @detiber on slack seems CAPA is almost ready to be moved to the test framework, after that happens we can probably delete this code. 

I'll close this for now and clear the milestone, feel free to reopen if I missed anything, thanks!

/close
/milestone Next

**Pull request closed without merging**